assets/src/stories-editor/blocks/amp-story-page/edit.js

```diff
@@ -88,6 +91,10 @@ const PageEdit = ( {
 			media: mediaObject,
 			videoFeaturedImage: videoThumbnail,
 			getBlockOrder,
+			getImmovableBlocks: ( pageClientId ) => {
```

Comment:
You only ever invoke this with `clientId` - why not actually extract the list here and store it in a `immovableBlocks` variable? Or is there a need for a function here? It also makes the dependencies of the subsequent `useEffect`'s more clear.

However, do remember to add `clientId` to the dependencies of this `useSelect` (it's already used here, so it's a mistake it hasn't been done already).

@swissspidy, do you know if we can add `useSelect` to the automatic eslint-based dependency-check? There's probably an option to add custom hooks to the validator? This hook is probably missing more than just `clientId`.

Reply:
Changed in 0197f85.

Reply:
@barklund Good question! I always thought it's either a limitation in the lint rule or a bug in the `useSelect` implementation. Looking at [the code](https://github.com/facebook/react/blob/9e64bf18e11828d6b4c0363bff5ed2eca1ccd838/packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js#L28-L35), it seems like we can just pass a regex/list of additional hooks to check against. Will whip up a quick PR to test this!